pkg/scaffold/resource/options.go

```diff
@@ -0,0 +1,234 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/gobuffalo/flect"
+)
+
+const (
+	versionPattern = "^v\\d+(alpha\\d+|beta\\d+)?$"
+
+	groupRequired = "group cannot be empty"
+	versionRequired = "version cannot be empty"
+	kindRequired = "kind cannot be empty"
+)
+
+var versionRegex = regexp.MustCompile(versionPattern)
+
+// Options contains the information required to build a new Resource
+type Options struct {
+	// Namespaced is true if the resource is namespaced.
+	Namespaced bool
+
+	// Group is the API Group. Does not contain the domain.
+	Group string
+
+	// Version is the API version.
+	Version string
+
+	// Kind is the API Kind.
+	Kind string
+
+	// Plural is the API Kind plural form.
+	// Optional
+	Plural string
+
+	// Resource is the API Resource.
+	// Optional
+	Resource string
+
+	// CreateExampleReconcileBody will create a Deployment in the Reconcile example.
+	// v1 only
+	CreateExampleReconcileBody bool
+}
+
+func (opts *Options) Validate() error {
+	// Check that the required flags did not get a flag as their value
+	// NOTE(adirio): We must do this for all the required flags first or we may
+	// output the wrong error as flags may seem to be missing because Cobra
+	// assigned them to another flag.
+	if strings.HasPrefix(opts.Group, "-") {
+		return fmt.Errorf(groupRequired)
+	}
+	if strings.HasPrefix(opts.Version, "-") {
+		return fmt.Errorf(versionRequired)
+	}
+	if strings.HasPrefix(opts.Kind, "-") {
+		return fmt.Errorf(kindRequired)
+	}
+	// Now we can check that all the required flags are not empty
+	if len(opts.Group) == 0 {
+		return fmt.Errorf(groupRequired)
+	}
+	if len(opts.Version) == 0 {
+		return fmt.Errorf(versionRequired)
+	}
+	if len(opts.Kind) == 0 {
+		return fmt.Errorf(kindRequired)
+	}
+
+	// Check if the Group has a valid DNS1123 subdomain value
+	if err := IsDNS1123Subdomain(opts.Group); err != nil {
+		return fmt.Errorf("group name is invalid: (%v)", err)
+	}
+
+	// Check if the version follows the valid pattern
+	if !versionRegex.MatchString(opts.Version) {
+		return fmt.Errorf(
+			"version must match %s (was %s)", versionPattern, opts.Version)
+	}
+
+	// Check if the Kind is PascalCase
+	if opts.Kind != flect.Pascalize(opts.Kind) {
+		return fmt.Errorf("kind must be PascalCase (expected %s was %s)", flect.Pascalize(opts.Kind), opts.Kind)
+	}
+
+	// TODO: validate plural and resource strings if provided
```

Comment:
I think it can be removed. 
Why would like to check it? What is your motivation?

Reply:
It is a bit outdated, yes, but plural may still require some validation. Remember that there is a `--plural` flag that you can set to use a custom plural so we may need to do some validations (like for example only allowing lowercased letters).